friktion/friktion/friktion_anchor/errors/custom.py

```diff
@@ -174,24 +173,90 @@ def __init__(self) -> None:
     msg = "invalid swap admin"
 
 
-class InvalidEd25519InstructionData(ProgramError):
+class UnableToLoadEd25519Instruction(ProgramError):
     def __init__(self) -> None:
-        super().__init__(6019, "")
+        super().__init__(6019, "unable to load instruction at currentIdx-1 position")
 
     code = 6019
+    name = "UnableToLoadEd25519Instruction"
+    msg = "unable to load instruction at currentIdx-1 position"
+
+
+class InvalidEd25519InstructionData(ProgramError):
+    def __init__(self) -> None:
+        super().__init__(6020, "invalid signature data")
+
+    code = 6020
     name = "InvalidEd25519InstructionData"
-    msg = ""
+    msg = "invalid signature data"
+
+
+class CounterpartyMismatch(ProgramError):
+    def __init__(self) -> None:
+        super().__init__(
+            6021,
+            "mismatch between swap_order couterparty and signed message counterparty",
+        )
+
+    code = 6021
+    name = "CounterpartyMismatch"
+    msg = "mismatch between swap_order couterparty and signed message counterparty"
 
 
 class OptionAndGiveMintDontMatch(ProgramError):
     def __init__(self) -> None:
-        super().__init__(6020, "option and give mint don't match")
+        super().__init__(6022, "option and give mint don't match")
 
-    code = 6020
+    code = 6022
     name = "OptionAndGiveMintDontMatch"
     msg = "option and give mint don't match"
 
 
+class DisabledInstruction(ProgramError):
+    def __init__(self) -> None:
+        super().__init__(6023, None)
+
+    code = 6023
+    name = "DisabledInstruction"
+    msg = None
```

Comment:
We can have a base class and simplify a lot of the code, e.g.

```suggestion

class GenericError(ProgramError):
    code = 0
    name = "GenericError"
    msg = "Generic error"
    
    def __init__(self):
        super().__init__(self.code, self.msg)
    
class DisabledInstruction(GenericError):
    code = 6023
    name = "DisabledInstruction"
    msg = None
```

Then also all the errors below and in the other file can inherit from the `GenericError`
and be derived without using the `__init__`.

Reply:
If you see my below commend, `friktion_anchor` and `inertia_anchor` are auto-generated directories, could we ignore these directories with the linter?

Reply:
Interesting. How are these generated?

Reply:
https://github.com/tradeparadigm/sdks#code-guidelines

but now you could do `pre-commit run --all-files`

Reply:
Note: we'll do it on top of your final changes